llvm/lib/Transforms/Vectorize/LoopVectorize.cpp

```diff
@@ -9144,6 +9144,15 @@ void LoopVectorizationPlanner::adjustRecipesForReductions(
             CurrentLinkI->getFastMathFlags());
         LinkVPBB->insert(FMulRecipe, CurrentLink->getIterator());
         VecOp = FMulRecipe;
+      } else if (PhiR->isInLoop() && Kind == RecurKind::Add &&
```

Comment:
Why is this still looking for `RecurdKind::Add` and why does this code still need to look at `CurrentLink->getOpcode()` ?